### PR TITLE
Implement marker clusters to declutter the map

### PR DESCRIPTION
**Pull request by sassypocoyo**
This is a substancial PR that implements marker clusters to help improve map overview by decluttering dense areas. This required the use of new libraries and changes to the next.config.js file, so I would like your review to make sure the project's stability won't be affected. Here's a list of the new libraries needed:

- **[leaflet.markercluster](https://leaflet.github.io/Leaflet.markercluster/)**
- **[react-leaflet-cluster](https://github.com/akursat/react-leaflet-cluster)**
- **[next-global-css](https://github.com/bem/next-global-css)** (Used to fix compatibility issues)

I think the use of this library is the best solution for this problem, it looks good and it has good interactivity and features.

image

Ps: I'm accepting better design ideas for the clusters (the styling is CSS based)

### CODE REVIEW

File: components/Map/style.css [hunk 1]
```diff
@@ -0,0 +1,13 @@
+.custom-marker-cluster {
```
Comment (nelsonmestevao) [inline components/Map/style.css:1]: Move this to `styles/global.css` and you won't need any extra dependencies :pray:

Comment (sassypocoyo) [reply components/Map/style.css:1]: Noted.

Comment (nelsonmestevao) [reply components/Map/style.css]: Remove the external depdency, please :pray:

Comment (sassypocoyo) [reply components/Map/style.css]: Ohh I got it now, we don't need next-global-css. It's literally in the name, global 😅 anyway, I'm on it :)

Comment (sassypocoyo) [reply components/Map/style.css]: > Ohh I got it now, we don't need next-global-css. It's literally in the name, global sweat_smile anyway, I'm on it :)

Update:

image

@nelsonmestevao

Comment (nelsonmestevao) [reply components/Map/style.css]: I guess we will to use global css until the new release happens :'(  (looking forward to https://github.com/yuzhva/react-leaflet-markercluster/pull/194)

**Review by nelsonmestevao — APPROVED** [review]

File: components/Map/style.css (hunk 1 above)
Comment (sassypocoyo) [reply components/Map/style.css]: Ahaha true, last commit was two years ago :')

**Pull request merged**